mxcubeweb/core/adapter/adapter_base.py

```diff
@@ -40,15 +43,23 @@ def __init__(self, ho, role, app):
         self._unique = True
         self._msg = ""
 
+    @classmethod
+    def can_adapt(cls, ho):
+        return any(isinstance(ho, t) for t in cls.SUPPORTED_TYPES)
+
+    def __init_subclass__(cls, **kwargs):
```

Comment:
Ive been thinking about how to achieve this for some time and `__init_subclass__` was the missing piece that allows for the registration of the actual existing sub classes so that we know which adapters that exist and in extension what hardware objects we support.